src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -237,13 +256,116 @@ public static class AkkaClusterHostingExtensions
             {
                 middle = middle.AddHocon($"akka.cluster.pub-sub = \"{role}\"");
             }
-                
+
             return middle.WithActors((system, registry) =>
             {
                 // force the initialization
                 var mediator = DistributedPubSub.Get(system).Mediator;
                 registry.TryRegister<DistributedPubSub>(mediator);
             });
         }
+
+        /// <summary>
+        /// Creates a new <see cref="ClusterSingletonManager"/> to host an actor created via <see cref="actorProps"/>.
+        ///
+        /// If <see cref="createProxyToo"/> is set to <c>true</c> then this method will also create a <see cref="ClusterSingletonProxy"/> that
+        /// will be added to the <see cref="ActorRegistry"/> using the key <see cref="TKey"/>. Otherwise this method will register nothing with
+        /// the <see cref="ActorRegistry"/>.
+        /// </summary>
+        /// <param name="builder">The builder instance being configured.</param>
+        /// <param name="singletonName">The name of this singleton instance. Will also be used in the <see cref="ActorPath"/> for the <see cref="ClusterSingletonManager"/> and
+        /// optionally, the <see cref="ClusterSingletonProxy"/> created by this method.</param>
+        /// <param name="actorProps">The underlying actor type. SHOULD NOT BE CREATED USING <see cref="ClusterSingletonManager.Props"/></param>
+        /// <param name="options">Optional. The set of options for configuring both the <see cref="ClusterSingletonManager"/> and
+        /// optionally, the <see cref="ClusterSingletonProxy"/>.</param>
+        /// <param name="createProxyToo">When set to <c>true></c>, creates a <see cref="ClusterSingletonProxy"/> that automatically points to the <see cref="ClusterSingletonManager"/> created by this method.</param>
```

Comment:
It's common for users to create both a `ClusterSingletonManager` and a `ClusterSingletonProxy` at the same time, so I've made that the default behavior. This method won't store the `ClusterSingletonManager` in the `ActorRegistry` because there's not much use in doing that - you can't interact with the `ClusterSingletonManager` directly - you need the  `ClusterSingletonProxy` for that. Hence why I've designed the method this way.